### PR TITLE
FAT2-158 - Move data importer function

### PR DESCRIPTION
**Pull request by nouriach**
- Data Importer Function brought from out of Courses
- Foreach brought in to loop through multiple URLs
- Test added to accommodate multiple URL endpoints

### CODE REVIEW

File: azure/template.json [hunk 1]
```diff
@@ -0,0 +1,374 @@
...
+                                "name": "AzureWebJobsStorage",
+                                "value": "[reference('storage-account').outputs.storageConnectionString.value]"
+                            },
+                            {
+                                "name": "APPINSIGHTS_INSTRUMENTATIONKEY",
+                                "value": "[reference('importer-function-app-insights').outputs.InstrumentationKey.value]"
+                            },
+                            {
+                                "name": "FUNCTIONS_EXTENSION_VERSION",
+                                "value": "~2"
```
Comment (dashton82) [inline azure/template.json:302]: This needs updating to v3

File: azure/template.json [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+                            }
+                        ]
+                    }
+                }
+            },
+            "dependsOn": [
+                "app-service-plan"
+            ]
+        },
+        {
```
Comment (dashton82) [inline azure/template.json:320]: This section can be removed I believe

File: azure/template.json [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+                    },
+                    "appServiceAppSettings": {
+                        "value": [
+                            {
+                                "name": "Environment",
+                                "value": "[parameters('environmentName')]"
+                            },
+                            {
+                                "name": "ConfigNames",
+                                "value": "SFA.DAS.DataLoader"
```
Comment (dashton82) [inline azure/template.json:198]: Check with Adam or Jack but i think this whole section can be removed

File: azure-pipelines.yml [hunk 4]
```diff
@@ -1 +1,106 @@
...
+        azure/**
+      TargetFolder: '$(build.artifactstagingdirectory)/publish'
+      OverWrite: true
+
+  - task: PublishBuildArtifacts@1
+    displayName: 'Publish Artifact'
+    inputs:
+      PathtoPublish: '$(build.artifactstagingdirectory)/publish'
+
+- job: 'DACPACBuild'
```
Comment (dashton82) [inline azure-pipelines.yml:86]: This can be removed, there is no database 😄

File: azure-pipelines.yml [hunk 5]
```diff
@@ -1 +1,84 @@
...
+      noCache: true
+
+  - task: DotNetCoreCLI@2
+    displayName: Build
+    inputs:
+      projects: 'src/**/*.csproj'
+      arguments: '--configuration $(buildConfiguration) --no-restore'
+      
+  - task: DotNetCoreCLI@2
+    displayName: Test
```
Comment (jack-education) [inline azure-pipelines.yml:50]:
```suggestion
    displayName: Unit Tests
```

File: azure/template.json [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        },
+        "serviceName": {
+            "type": "string"
+        },
+        "sharedEnvResourceGroup": {
+            "type": "string"
+        },
+        "sharedBackEndAppServicePlanName": {
+            "type": "string"
+        },
```
Comment (jack-education) [inline azure/template.json:19]: Not needed

File: azure-pipelines.yml [hunk 7]
```diff
@@ -1 +1,84 @@
...
+    type: github
+    name: SkillsFundingAgency/das-platform-building-blocks
+    ref: refs/heads/master
+    endpoint: "GitHub (SFA)"
+
+jobs:
+- job: 'CodeBuild'
+  pool:
+    name: 'DAS - Continuous Integration'
+    demands: Agent.OS -equals Linux
```
Comment (jack-education) [inline azure-pipelines.yml:26]:
```suggestion
```
We can remove the demand, it can run on Windows/Linux

File: azure-pipelines.yml [hunk 8]
```diff
@@ -1 +1,84 @@
...
+  workspace:
+    clean: all  
+  steps:
+  - task: gittools.gitversion-preview.gitversion-task.GitVersion@5
```
Comment (jack-education) [inline azure-pipelines.yml:30]:
```suggestion
  - task: UseGitVersion@5
```
Latest version of task that is installed on the organisation

File: azure/template.json [hunk 9]
```diff
@@ -0,0 +1,191 @@
...
+                    "appInsightsName": {
+                        "value": "[variables('importerFunctionAppName')]"
+                    },
+                    "attachedService": {
+                        "value": "[variables('importerFunctionAppName')]"
+                    }
+                }
+            }
+        },
+        {ropertie
```
Comment (jack-education) [inline azure/template.json:121]:
```suggestion
        {
```

File: azure/template.json [hunk 10]
```diff
@@ -0,0 +1,191 @@
...
+                "parameters": {
+                    "functionAppName": {
+                        "value": "[variables('importerFunctionAppName')]"
+                    },
+                    "appServicePlanName": {
+                        "value": "[variables('appServicePlanName')]"
+                    },
+                    "appServicePlanResourceGroup": {
+                        "value": "[resourceGroup().name]"
+                    },
```
Comment (jack-education) [inline azure/template.json:140]:
```suggestion
                    },
                    "subnetResourceId": {
                        "value": "[reference('function-app-subnet').outputs.SubnetResourceId.value]"
                    },
```

File: azure/template.json [hunk 11]
```diff
@@ -0,0 +1,191 @@
...
+        "appServicePlanSize": {
+            "type": "string",
+            "defaultValue": "1"
+        },
+        "loggingRedisConnectionString": {
+            "type": "string"
+        },
+        "configurationStorageConnectionString": {
+            "type": "securestring"
+        },
```
Comment (jack-education) [inline azure/template.json:27]:
```suggestion
        },
        "sharedStorageAccountConnectionString": {
            "type": "securestring"
        },
```

File: azure/template.json [hunk 12]
```diff
@@ -0,0 +1,191 @@
...
+                                "name": "EnvironmentName",
+                                "value": "[parameters('environmentName')]"
+                            },
+                            {
+                                "name": "Version",
+                                "value": "1.0"
+                            },
+                            {
+                                "name": "AzureWebJobsStorage",
+                                "value": "[reference('storage-account').outputs.storageConnectionString.value]"
```
Comment (jack-education) [inline azure/template.json:161]:
```suggestion
                                "value": "[parameters('sharedStorageAccountConnectionString')]"
```

File: azure/template.json (hunk 9 above)
Comment (jack-education) [reply azure/template.json:121]: 'ropertie' needs to be removed

File: azure-pipelines.yml [hunk 13]
```diff
@@ -1 +1,83 @@
...
+- job: 'CodeBuild'
+  pool:
+    name: 'DAS - Continuous Integration'
+  workspace:
+    clean: all  
+  steps:
+  - task: UseGitVersion@5
+    displayName: GitVersion
+    inputs:
+      configFilePath: GitVersion.yml
```
Comment (jack-education) [inline azure-pipelines.yml:32]:
```suggestion
      configFilePath: GitVersion.yml
      versionSpec: 5.x
```

File: azure/template.json [hunk 14]
```diff
@@ -0,0 +1,194 @@
...
+        },
+        "loggingRedisConnectionString": {
+            "type": "string"
+        },
+        "configurationStorageConnectionString": {
+            "type": "securestring"
+        },
+        "sharedStorageAccountConnectionString": {
+            "type": "securestring"
+        },
```
Comment (jack-education) [inline azure/template.json:30]:
```suggestion
        },
        "sharedEnvVirtualNetworkName": {
            "type": "string"
        },
```

File: azure/template.json [hunk 15]
```diff
@@ -0,0 +1,197 @@
...
+    "parameters": {
+        "resourceEnvironmentName": {
+            "type": "string"
+        },
+        "environmentName": {
+            "type": "string"
+        },
+        "serviceName": {
+            "type": "string"
+        },
```
Comment (jack-education) [inline azure/template.json:13]:
```suggestion
        },
        "sharedEnvResourceGroup": {
            "type": "string"
        },
```

**Pull request merged**